### ISSUE TITLE
Minifiing he.js breaks Safari

### ISSUE THREAD

**Issue by klimashkin** [terser/terser#729]
Hi!

If I include [he.js](https://github.com/mathiasbynens/he) in the bundle and use the latest terser with it (terser 4.7.0, terser-webpack-plugin 3.0.3), then Safari 13.1.1 breaks on this:
`SyntaxError: Invalid character '\ud835'`

<img width="595" alt="Screen Shot 2020-06-11 at 5 09 06 PM" src="https://user-images.githubusercontent.com/576077/84450876-5095d780-ac06-11ea-9eb4-d119f77aec8e.png">

I think it's this piece of code:
https://github.com/mathiasbynens/he/blob/master/data/encode-map.json#L997

Here is the config:
```js
terserOptions: {
  ecma: 5,
  keep_classnames: false,
  keep_fnames: false,
  ie8: false,
  nameCache: null,
  safari10: false,
  toplevel: false,
  warnings: false,
  module: false,

  parse: {
    bare_returns: false,
    ecma: 2019,
    html5_comments: true,
    shebang: true,
  },
  compress: {
    arrows: false,
    arguments: true,
    booleans: true,
    booleans_as_integers: false,
    collapse_vars: true,
    comparisons: true,
    computed_props: true,
    conditionals: true,
    dead_code: true,
    defaults: true,
    directives: true,
    drop_console: false,
    drop_debugger: true,
    ecma: isModern ? 2019 : 5,
    evaluate: true,
    expression: false,
    global_defs: {},
    hoist_funs: false,
    hoist_props: true,
    hoist_vars: false,
    if_return: true,
    inline: false,
    join_vars: false,
    keep_classnames: false,
    keep_fargs: true,
    keep_fnames: false,
    keep_infinity: true,
    loops: true,
    module: false,
    negate_iife: true,
    passes: 1,
    properties: true,
    pure_funcs: null,
    pure_getters: 'strict',
    reduce_funcs: true,
    reduce_vars: true,
    sequences: true,
    side_effects: true,
    switches: true,
    toplevel: false,
    top_retain: null,
    typeofs: true,
    unsafe: false,
    unsafe_arrows: false,
    unsafe_comps: false,
    unsafe_Function: false,
    unsafe_math: false,
    unsafe_methods: false,
    unsafe_proto: false,
    unsafe_regexp: false,
    unsafe_undefined: false,
    unused: true,
    warnings: true,
  },
  mangle: {
    eval: false,
    keep_classnames: false,
    keep_fnames: false,
    module: false,
    reserved: [],
    toplevel: false,
    safari10: true,
  },
  output: {
    ascii_only: false,
    beautify: true,
    braces: false,
    comments: false,
    ecma: isModern ? 6 : 5,
    indent_level: 2,
    indent_start: 0,
    inline_script: false,
    keep_numbers: false,
    keep_quoted_props: false,
    max_line_len: 400, // To make devtools responsive, should not be very small to avoid increasing size too much
    preamble: null,
    quote_keys: false,
    quote_style: 3,
    safari10: false,
    semicolons: true,
    shebang: true,
    webkit: false,
    width: 120,
    wrap_iife: false,
  },
},
```

**Comment by fabiosantoscode** [comment]
I can't believe this :/

I just checked it on my webkit-based browser locally.

You can work around this issue by setting `isModern` to false, or setting `output.keep_quoted_props` to true.

**Comment by JLHwung** [comment]
See the second test of https://kangax.github.io/compat-table/es6/#test-Updated_identifier_syntax: Safari 13 does not support non-BMP identifier.

It is fixed in Safari 14 though.

**Comment by Mike-Benoit** [comment]
Similar issue appears with Leaflet too: Invalid character '\u00b7'.

**Comment by fabiosantoscode** [comment]
Closed in d51b6b9c22e8e8452a7ca0bb27b0f80b54b8d415
